examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,290 @@
-        print("\tF1: %1.3f\n" % f1_score(y_test, y_pred))
+        Returns
+        -------
+        proba : ndarray of shape (n_samples, 2)
+            Predicted probabilities of the 2 classes.
+        """
+        df = self.decision_function(X)
+        calibrated_df = (df - self.df_min_) / (self.df_max_ - self.df_min_)
+        proba_class1 = np.clip(calibrated_df, 0, 1)
+        proba_class0 = 1 - proba_class1
+        proba = np.c_[proba_class0, proba_class1]
+        return proba
 
-        fraction_of_positives, mean_predicted_value = \
-            calibration_curve(y_test, prob_pos, n_bins=10)
 
-        ax1.plot(mean_predicted_value, fraction_of_positives, "s-",
-                 label="%s (%1.3f)" % (name, clf_score))
+lr = LogisticRegression(C=1.)
+svc = NaivelyCalibratedLinearSVC(max_iter=10000)
+svc_isotonic = CalibratedClassifierCV(svc, cv=2, method='isotonic')
+svc_sigmoid = CalibratedClassifierCV(svc, cv=2, method='sigmoid')
 
-        ax2.hist(prob_pos, range=(0, 1), bins=10, label=name,
-                 histtype="step", lw=2)
+clf_list = [(lr, 'Logistic'),
+            (svc, 'SVC'),
+            (svc_isotonic, 'SVC + Isotonic'),
+            (svc_sigmoid, 'SVC + Sigmoid')]
 
-    ax1.set_ylabel("Fraction of positives")
-    ax1.set_ylim([-0.05, 1.05])
-    ax1.legend(loc="lower right")
-    ax1.set_title('Calibration plots  (reliability curve)')
+fig = plt.figure(figsize=(10, 10))
+gs = GridSpec(4, 2)
 
-    ax2.set_xlabel("Mean predicted value")
-    ax2.set_ylabel("Count")
-    ax2.legend(loc="upper center", ncol=2)
+ax1 = fig.add_subplot(gs[:2, :2])
+viz_objects = {}
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    viz = plot_calibration_curve(
+        clf, X_test, y_test, n_bins=10, name=name, ax=ax1, color=colors(i)
+    )
+    viz_objects[name] = viz
 
-    plt.tight_layout()
+ax1.grid()
+ax1.set_title('Calibration plots (SVC)')
 
-# Plot calibration curve for Gaussian Naive Bayes
-plot_calibration_curve(GaussianNB(), "Naive Bayes", 1)
+# Add histogram
+grid_positions = [(2, 0), (2, 1), (3, 0), (3, 1)]
+for i, (_, name) in enumerate(clf_list):
+    row, col = grid_positions[i]
+    ax = fig.add_subplot(gs[row, col])
 
-# Plot calibration curve for Linear SVC
-plot_calibration_curve(LinearSVC(max_iter=10000), "SVC", 2)
+    ax.hist(
+        viz_objects[name].y_prob, range=(0, 1), bins=10, label=name,
+        color=colors(i)
+    )
+    ax.set(title=name, xlabel="Mean predicted probability", ylabel="Count")
 
+plt.tight_layout()
 plt.show()
+
+# %%
+# :class:`~sklearn.svm.LinearSVC` shows the opposite
+# behavior to :class:`~sklearn.naive_bayes.GaussianNB`; the calibration
+# curve has a sigmoid shape, which is typical for an under-confident
+# classifier. In the case of :class:`~sklearn.svm.LinearSVC`, this is caused
+# by the margin property of the hinge loss, which lets the model focus on
+# difficult samples that are close to the decision boundary (the support
+# vectors).
```

Comment:
I am looking into this, and your point below. The [Niculescu-Mizil & Caruana paper](https://www.cs.cornell.edu/~alexn/papers/calibration.icml05.crc.rev3.pdf) states this but does not expand more. I am trying to find more information on this but I just see other people stating the same thing and quoting the paper but not expanding more!

Reply:
Here is my intuition: compared to the logistic loss, the hinge loss is not impacted by examples far away from the decision boundary (samples that are not active support vectors at the optimum of the empirical risk): so it would make sense for the hinge loss to not try to separate samples in the high confidence regions leading to flatter calibration curves near the 0 and 1 bounds (assuming naive linear / min-max scaling of the decision function on the x axis of the calibration curve). This would lead to a sigmoid shape for the calibration curve hence under confidence.

Reply:
Maybe you could link to the  Niculescu-Mizil & Caruana paper to back-up the analysis of this example (even if it does not give more details).